daemon/containerd/image_pull.go

```diff
@@ -73,8 +73,14 @@ func (i *ImageService) PullImage(ctx context.Context, image, tagOrDigest string,
 	finishProgress := jobs.showProgress(ctx, out, pp)
 	defer finishProgress()
 
+	var sentPullingFrom bool
 	ah := images.HandlerFunc(func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
 		if images.IsManifestType(desc.MediaType) {
+			if !sentPullingFrom {
```

Comment:
would this code always be executed (also if the manifest happens to be cached already)?

Reply:
Yup

Reply:
Thanks for checking!

Reply:
Fixed the unfinished downloading message: https://github.com/moby/moby/pull/46515